proj/src/main/webapp/chat.js

```diff
@@ -1035,12 +1035,53 @@ function addUserInfoToDom(userObj) {
     }
 }
 
+$( "#btnChangePassword" ).click(function() {
+    initChangePasswordButton();
+});
+
+function initChangePasswordButton(){
+    const auth = firebase.auth();
+    const user = auth.currentUser;
+    const email = user.email;
+    var oldPassword = document.getElementById("oldPass").value;
+    var newPassword = document.getElementById("newPass").value;
+    const promise = auth.signInWithEmailAndPassword(email, oldPassword).then(function(){
+        if (meetPWRequirements(newPassword)) {
+            user.updatePassword(newPassword).then(function() {
+                alert("Password changed successfully");
+            }).catch(function(error) {
+                alert(error.message);
+            });
+        }
+    });
+    // Catches errors for old passwords
+    promise.catch(error => alert(error.message));
+}
+
+function meetPWRequirements(password){
+    errors = [];
+    if (password.length < 6) {
+        errors.push("Your password must contain at least 6 characters."); 
+    }
+    if (password.search(/[0-9]/) < 0) {
+        errors.push("Your password must contain at least one number."); 
+    }
+    if (password.search(/[$-/:-?{-~!"^_`@#\[\]\\]/) < 0) {
+        errors.push("Your password must contain at least one symbol."); 
+    }
+    if (errors.length > 0) {
+        // Shows what the new password is missing, if any
+        alert(errors.join("\n"));
+        return false;
+    }
+    return true;
+}
+
```

Comment:
Rather than recoding the logic for verifying the password, I would reuse the logic that is in signup. However I think signup imports chat so you would have a circular dependency. Is it possible to redirect them to a page similar to sign up or maybe move the password validation things to its own module separate from signup similar to how notifications and locations does things so it can be brought into both chat and signup?

Reply:
i thought about that and was considering importing just the meetRequirements function but that only return a boolean. i felt the validation function was too small for a separate module because the one in signup is more focused on triggering styling changes on each key entered while this is basic version for a quick password change. lmk what you think

Reply:
I still think it's better to reuse existing code. That way if we decide the password should be 7 letters instead of 6, we won't accidentally forget to change it in one of the places. Or if we notice the regex is missing a character in one of the places we won't forget to update both places.